lib/Runtime/Provisioner/Provisioner.cpp

```diff
@@ -63,11 +68,11 @@ Provisioner::provision(std::vector<std::unique_ptr<DAGNode>> &networks,
     // insufficient space.
 
     // Set backend to match the device.
-    backend_.reset(createBackend(currDevice->second->getBackendKind()));
+    backend_.reset(createBackend(devices_[currDevice_]->getBackendKind()));
```

Comment:
nit: Rather than do the look up repeatedly, can we store `auto* device = devices_[currentDevice_]`and reference that?